### PR TITLE
Update Certificate Authentication to 6.0

### LINKED ISSUE #23229: .NET 6 and Add .NET 6 xref APIs when they're published
See #23226

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 35b7960d-6de5-78ee-bc45-d7218239bbc8
* Version Independent ID: d2db534b-198a-dc99-5b5a-0525582d39ac
* Content: [Configure certificate authentication in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/certauth?view=aspnetcore-5.0)
* Content Source: [aspnetcore/security/authentication/certauth.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/security/authentication/certauth.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @blowdart
* Microsoft Alias: **bdorrans**

### PR DESCRIPTION
**Pull request by serpent5**
Fixes #23229.

[Internal Preview URL](https://review.docs.microsoft.com/en-us/aspnet/core/security/authentication/certauth?view=aspnetcore-6.0&branch=pr-en-us-24550).

I've created a 6.0 sample project and moved all inline code into there. I've left the content alone as much as I can, changing only references to `Startup` and updating the code for 6.0. I've also dropped the first section that discusses using `HttpClient` directly, as I think the section that follows covers the same details without having to worry about the lifetime of `HttpClient` et al.

### CODE REVIEW

File: aspnetcore/security/authentication/certauth.md [hunk 1]
```diff
@@ -29,33 +29,14 @@ An alternative to certificate authentication in environments where proxies and l
...
+`builder.Services.AddAuthentication(CertificateAuthenticationDefaults.AuthenticationScheme).AddCertificate(...);` with your options, providing a delegate for `OnCertificateValidated` to do any supplementary validation on the client certificate sent with requests. Turn that information into a `ClaimsPrincipal` and set it on the `context.Principal` property.
 
 If authentication fails, this handler returns a `403 (Forbidden)` response rather a `401 (Unauthorized)`, as you might expect. The reasoning is that the authentication should happen during the initial TLS connection. By the time it reaches the handler, it's too late. There's no way to upgrade the connection from an anonymous connection to one with a certificate.
 
-Also add `app.UseAuthentication();` in the `Startup.Configure` method. Otherwise, the `HttpContext.User` will not be set to `ClaimsPrincipal` created from the certificate. For example:
+Also add `app.UseAuthentication();` in *Program.cs*. Otherwise, the `HttpContext.User` will not be set to `ClaimsPrincipal` created from the certificate. For example:
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/certauth.md:37]:
```suggestion
`UseAuthentication` is required to set  `HttpContext.User` to `ClaimsPrincipal` created from the certificate. For example:
```

File: aspnetcore/security/authentication/certauth.md [hunk 2]
```diff
@@ -29,33 +29,14 @@ An alternative to certificate authentication in environments where proxies and l
...
 
-In your web app, add a reference to the [Microsoft.AspNetCore.Authentication.Certificate](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.Certificate) package. Then in the `Startup.ConfigureServices` method, call
-`services.AddAuthentication(CertificateAuthenticationDefaults.AuthenticationScheme).AddCertificate(...);` with your options, providing a delegate for `OnCertificateValidated` to do any supplementary validation on the client certificate sent with requests. Turn that information into a `ClaimsPrincipal` and set it on the `context.Principal` property.
+In your web app, add a reference to the [Microsoft.AspNetCore.Authentication.Certificate](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.Certificate) package. Then in *Program.cs*, call
+`builder.Services.AddAuthentication(CertificateAuthenticationDefaults.AuthenticationScheme).AddCertificate(...);` with your options, providing a delegate for `OnCertificateValidated` to do any supplementary validation on the client certificate sent with requests. Turn that information into a `ClaimsPrincipal` and set it on the `context.Principal` property.
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/certauth.md:33]: Dead wood such as "and then, then in, Also" are super pet peeves.

Comment (Rick-Anderson) [reply aspnetcore/security/authentication/certauth.md:33]:
```suggestion
In the web app:

* Add a reference to the [Microsoft.AspNetCore.Authentication.Certificate](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.Certificate) package.
* In *Program.cs*, call
`builder.Services.AddAuthentication(CertificateAuthenticationDefaults.AuthenticationScheme).AddCertificate(...);`. Provide a delegate for `OnCertificateValidated` to do any supplementary validation on the client certificate sent with requests. Turn that information into a `ClaimsPrincipal` and set it on the `context.Principal` property.
```

File: aspnetcore/security/authentication/certauth.md [hunk 3]
```diff
@@ -110,84 +91,15 @@ The handler has two events:
...
-                }
-            };
-        });
-    ```
+    :::code language="csharp" source="certauth/samples/6.x/CertAuthSample/Snippets/Program.cs" id="snippet_AddCertificateOnCertificateValidated":::
 
 If you find the inbound certificate doesn't meet your extra validation, call `context.Fail("failure reason")` with a failure reason.
 
-For real functionality, you'll probably want to call a service registered in dependency injection that connects to a database or other type of user store. Access your service by using the context passed into your delegate. Consider the following example in `Startup.ConfigureServices`:
+For real functionality, you'll probably want to call a service registered in dependency injection that connects to a database or other type of user store. Access your service by using the context passed into your delegate. Consider the following example:
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/certauth.md:100]:
```suggestion
For better functionality, call a service registered in dependency injection that connects to a database or other type of user store. Access the service by using the context passed into the delegate. Consider the following example:
```

**Review by Rick-Anderson — APPROVED** [review]
> I've left the content alone as much as I can, changing only references t
Couldn't resist some cleanup. Much better. For non beginner documents we probably don't even need to mention the code goes in *Progam.cs*, as that's obvious.

**Review by wadepickett — APPROVED** [review]
Correct for Rick's suggestions and looks good for what was edited for the PR.  There are nits with the original content here and there, but I stuck with the edits.

**Comment by serpent5** [comment]
> Couldn't resist some cleanup.

😆 I don't blame you. Those were some great suggestions, thanks.

**Comment by wadepickett** [comment]
.. and great work!  :)

**Pull request merged**